File: java/src/main/java/industries/atende/ts/driver/Response.java
/**
 * Copyright 2025 Atende Industries
 */
package industries.atende.ts.driver;

import java.util.Objects;

/**
 * @class Response
 * @brief Represents the result of an operation, encapsulating a status.
 *
 * Used as a base response type for various operations, indicating success, failure,
 * or other outcome as represented by {@link ResponseStatus}.
 */
public class Response {

    /** @brief The status of the response. */
    private final ResponseStatus status;

    /**
     * @brief Constructs a new Response with the specified status.
     *
     * @param status The {@link ResponseStatus} of the operation. Must not be null.
     * @throws NullPointerException if the status is null.
     */
    Response(ResponseStatus status) {
        this.status = Objects.requireNonNull(status, "Status must not be null");
    }

    /**
     * @brief Returns the status of the response.
     *
     * @return The {@link ResponseStatus} associated with this response.
     */
    public ResponseStatus getStatus() {
        return status;
    }
}

File: csharp/TStorage/Interfaces/Response.cs
/*
 * Copyright 2025 Atende Industries
 */

namespace TStorage.Interfaces
{
    /// <summary>
    /// Represents the result of an operation, encapsulating a status.
    /// Used as a base response type for various operations, indicating success, failure,
    /// or other outcome as represented by <see cref="ResponseStatus"/>
    /// </summary>
    public class Response
    {
        /// <summary>
        /// Initializes a new instance of the Response.
        /// </summary>
        /// <param name="status"> Result of the response status. </param>
        public Response(ResponseStatus status)
        {
            Status = status;
        }

        /// <summary> The status of the response. </summary>
        public readonly ResponseStatus Status;
    }
}